modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ModuleFlowProcessingTemplate.java

```diff
@@ -51,6 +56,11 @@ public Message getMessage() {
     return message;
   }
 
+  @Override
+  public Optional<SourceNotification> getSourceNotification() {
+    return ofNullable(sourceNotification);
```

Comment:
same thing. What if i wnat to fire many?

Reply:
I think they should accumulate and all be fired. But again, this is because of the `Event` issue.
If you don't want it to be associated to an Event. I think we should have a `fire` and a `fireOnEvent` for sources.

Reply:
The problem is that `fire` would have to fire something different than an `ExtensionNotification`